src/main/java/com/zilen/weather/service/WeatherService.java

```diff
@@ -14,15 +16,26 @@
     private final String appid;
     private final String url;
     private final RestTemplate restTemplate;
+    private final WeatherRepository weatherRepository;
 
-    public WeatherService(@Value("${com.zilen.appid}") String appid, @Value("${com.zilen.url}") String url, RestTemplate restTemplate) {
+    public WeatherService(@Value("${com.zilen.appid}") String appid, @Value("${com.zilen.url}") String url, RestTemplate restTemplate, WeatherRepository weatherRepository) {
         this.appid = appid;
         this.url = url;
         this.restTemplate = restTemplate;
+        this.weatherRepository = weatherRepository;
     }
 
     public WeatherDTO findByCityName(String cityName) {
         ResponseEntity<WeatherDTO> response = restTemplate.getForEntity(url + cityName + "&units=metric&appid=" + appid, WeatherDTO.class);
+
+        Weather weather = new Weather();
```

Comment:
It would be better to create a class that will be responsible for conversions between entity and dto.